Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM_2nd.cpp

```diff
@@ -0,0 +1,540 @@
+            Array4<Real> const& M_error_yface = Mfield_error[1]->array(mfi);
+            Array4<Real> const& M_error_zface = Mfield_error[2]->array(mfi);
+            Array4<Real> const& a_temp_xface = a_temp[0]->array(mfi);
+            Array4<Real> const& a_temp_yface = a_temp[1]->array(mfi);
+            Array4<Real> const& a_temp_zface = a_temp[2]->array(mfi);
+            Array4<Real> const& a_temp_static_xface = a_temp_static[0]->array(mfi);
+            Array4<Real> const& a_temp_static_yface = a_temp_static[1]->array(mfi);
+            Array4<Real> const& a_temp_static_zface = a_temp_static[2]->array(mfi);
+            Array4<Real> const& b_temp_xface= b_temp[0]->array(mfi);
+            Array4<Real> const& b_temp_yface= b_temp[1]->array(mfi);
+            Array4<Real> const& b_temp_zface= b_temp[2]->array(mfi);
+
+            // extract stencil coefficients
+            Real const * const AMREX_RESTRICT coefs_x = m_stencil_coefs_x.dataPtr();
+            int const n_coefs_x = m_stencil_coefs_x.size();
+            Real const * const AMREX_RESTRICT coefs_y = m_stencil_coefs_y.dataPtr();
+            int const n_coefs_y = m_stencil_coefs_y.size();
+            Real const * const AMREX_RESTRICT coefs_z = m_stencil_coefs_z.dataPtr();
+            int const n_coefs_z = m_stencil_coefs_z.size();
+
+            // extract tileboxes for which to loop
+            Box const& tbx = mfi.tilebox(Bfield[0]->ixType().toIntVect()); /* just define which grid type */
+            Box const& tby = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+            Box const& tbz = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+            // reset the value of the M_iter_maxerror
+            M_iter_maxerror = -1.0;
+
+            // loop over cells and update fields
+            amrex::ParallelFor(tbx, tby, tbz,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+              // when working on M_xface(i,j,k, 0:2) we have direct access to M_xface(i,j,k,0:2) and Hx(i,j,k)
+              // Hy and Hz can be acquired by interpolation
+              // H_maxwell
+              Real Hx_xface = MacroscopicProperties::getH_Maxwell(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(1,0,0), Bx, M_prev_xface);
+              Real Hy_xface = MacroscopicProperties::getH_Maxwell(i, j, k, 1, amrex::IntVect(0,1,0), amrex::IntVect(1,0,0), By, M_prev_xface);
+              Real Hz_xface = MacroscopicProperties::getH_Maxwell(i, j, k, 2, amrex::IntVect(0,0,1), amrex::IntVect(1,0,0), Bz, M_prev_xface);
+              // H_bias
+              Real Hx_bias_xface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(1,0,0), amrex::IntVect(1,0,0), Hx_bias);
+              Real Hy_bias_xface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,1,0), amrex::IntVect(1,0,0), Hy_bias);
+              Real Hz_bias_xface = MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0,0,1), amrex::IntVect(1,0,0), Hz_bias);
```

Comment:
These lines seem redundant as well.

Reply:
Yes, there are some lines that are redundant now. But we will use them when we couple the LLG equation with the Maxwell equation. I will delete the lines if they are not involved in the coupling scheme.

Reply:
H_bias is not affected by the Maxwell scheme. these will be redundant anyways.

Reply:
You are right. 

H_bias is not affected by the Maxwell scheme. That is why they are calculated from the average. 

Re-check it to see if they really need to be deleted.